secure_storage/host/main.c

```diff
@@ -0,0 +1,258 @@
+			char *data, size_t data_len)
+{
+	TEEC_Operation op;
+	uint32_t origin;
+	TEEC_Result res;
+	size_t id_len = strlen(id);
+
+	memset(&op, 0, sizeof(op));
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_NONE, TEEC_NONE);
+
+	op.params[0].tmpref.buffer = id;
+	op.params[0].tmpref.size = id_len;
+
+	op.params[1].tmpref.buffer = data;
+	op.params[1].tmpref.size = data_len;
+
+	res = TEEC_InvokeCommand(&ctx->sess, TA_SECURE_STORAGE_CMD_WRITE_RAW,
+				 &op, &origin);
+	if (res != TEEC_SUCCESS)
+		printf("Command WRITE_RAW failed: 0x%x / %u\n", res, origin);
+
+	switch (res) {
+	case TEEC_SUCCESS:
+		break;
+	default:
+		printf("Command WRITE_RAW failed: 0x%x / %u\n", res, origin);
+	}
+
+	return res;
+}
+
+TEEC_Result delete_secure_object(struct test_ctx *ctx, char *id)
+{
+	TEEC_Operation op;
+	uint32_t origin;
+	TEEC_Result res;
+	size_t id_len = strlen(id);
+
+	memset(&op, 0, sizeof(op));
+	op.paramTypes = TEEC_PARAM_TYPES(TEEC_MEMREF_TEMP_INPUT,
+					 TEEC_NONE, TEEC_NONE, TEEC_NONE);
+
+	op.params[0].tmpref.buffer = id;
+	op.params[0].tmpref.size = id_len;
+
+	res = TEEC_InvokeCommand(&ctx->sess, TA_SECURE_STORAGE_CMD_DELETE,
+				 &op, &origin);
+
+	switch (res) {
+	case TEEC_SUCCESS:
+	case TEEC_ERROR_ITEM_NOT_FOUND:
+		break;
+	default:
+		printf("Command DELETE failed: 0x%x / %u\n", res, origin);
+	}
+
+	return res;
+}
+
+#define TEST_OBJECT_SIZE	7000
+
+int main(int argc, char *argv[])
+{
+	struct test_ctx ctx;
+	char obj1_id[] = "object#1";
+	char obj2_id[] = "object#2";
+	char obj1_data[TEST_OBJECT_SIZE];
+	char read_data[TEST_OBJECT_SIZE];
+	TEEC_Result res;
+
+	printf("Prepare session with the TA\n");
+	prepare_tee_session(&ctx);
+
+	/*
+	 * Create object, read it, delete it.
+	 */
+	printf("\nTest on object \"%s\"\n", obj1_id);
+
+	printf("- Create and load object in the TA secure storage\n");
+
+	memset(obj1_data, 0xA1, sizeof(obj1_data));
+
+	res = write_secure_object(&ctx, obj1_id,
+				  obj1_data, sizeof(obj1_data));
+	if (res != TEEC_SUCCESS)
+		errx(1, "Failed to create an object in the secure storage");
+
+	printf("- Read back the but in a too small buffer (should fail)\n");
+
+	res = read_secure_object(&ctx, obj1_id,
+				 read_data, sizeof(read_data) - 100);
+	if (res != TEEC_ERROR_SHORT_BUFFER)
+		errx(1, "Expected 'short buffer' error but got 0x%x\n", res);
+
+	printf("- Read back the object (into a well sized buffer)\n");
+
+	res = read_secure_object(&ctx, obj1_id,
+				 read_data, sizeof(read_data));
+	if (res != TEEC_SUCCESS)
+		errx(1, "Failed to create an object in the secure storage");
+	if (memcmp(obj1_data, read_data, sizeof(obj1_data)))
+		errx(1, "Unexpected content found in secure storage");
+
+	printf("- Delete the object\n");
+
+	res = delete_secure_object(&ctx, obj1_id);
+	if (res != TEEC_SUCCESS)
+		errx(1, "Failed to delete the object: 0x%x", res);
+
+	printf("- Try to access the object (should fail)\n");
```

Comment:
Same here, I would skip this "should fail" section.

Reply:
ok